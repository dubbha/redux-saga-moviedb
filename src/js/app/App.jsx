import React from 'react';
import { Switch, Route } from 'react-router-dom';
import Loadable from 'react-loadable';
import Loading from '../common/Loading';
import './style.sass';

const Search = Loadable({
  loader: () => import(/* webpackChunkName: "search" */ '../search'),
  loading: Loading,
});

const Film = Loadable({
  loader: () => import(/* webpackChunkName: "film" */ '../film'),
  loading: Loading,
});

const App = () => (
  <div className="app">
    <Switch>
      <Route path="/search/:by?/:query?" component={Search} />
      <Route path="/film/:id/:title?" component={Film} />
      <Route path="/" exact component={Search} />
    </Switch>
  </div>
);

export default App;
