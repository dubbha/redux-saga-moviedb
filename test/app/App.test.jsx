import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import App from 'app/App.jsx';

jest.mock('react', () => {
  const reactMock = require('React');
  reactMock.StrictMode = function StrictMode() {}; // named shallow mock
  return reactMock;
});

describe('app/App', () => {
  it('should render successfully', () => {
    const wrapper = shallow(<App />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
