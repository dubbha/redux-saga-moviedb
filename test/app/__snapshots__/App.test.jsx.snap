// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`app/App should render successfully 1`] = `
<div
  className="app"
>
  <Switch>
    <Route
      component={[Function]}
      path="/search/:by?/:query?"
    />
    <Route
      component={[Function]}
      path="/film/:id/:title?"
    />
    <Route
      component={[Function]}
      exact={true}
      path="/"
    />
  </Switch>
</div>
`;
